apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/ChunkingTests.cs

```diff
@@ -48,78 +49,63 @@ public ChunkingTests(TestDatabaseFixture testFixture, ITestOutputHelper outputHe
-            var size = _random.Next(10) + 10;
 
-            for (int i = 0; i < size; i++)
-            {
-                list.Add(GetLargeString(100000, 200000));
-            }
-            return list.ToArray();
-        }
+        private string[] GetLargeStringArray() =>
+            Enumerable.Range(0, _random.Next(10, 20)).Select(_ => GetLargeString(100000, 200000)).ToArray();
+
 
         private byte[] GetLargeBytes(int minSize, int maxSize)
         {
-            var size = _random.Next(maxSize - minSize) + minSize;
-            byte[] bytes = new byte[size];
-            for (int i = 0; i < size; i++)
-            {
-                bytes[i] = (byte)_random.Next(256);
-            }
+            int size = _random.Next(minSize, maxSize);
+            var bytes = new byte[size];
+            _random.NextBytes(bytes);
             return bytes;
         }
 
-        private byte[][] GetLargeBytesArray()
-        {
-            List<byte[]> list = new List<byte[]>();
-            var size = _random.Next(10) + 10;
-
-            for (int i = 0; i < size; i++)
-            {
-                list.Add(GetLargeBytes(100000, 200000));
-            }
-            return list.ToArray();
-        }
+        private byte[][] GetLargeBytesArray() =>
+            Enumerable.Range(0, _random.Next(10, 20)).Select(_ => GetLargeBytes(100000, 200000)).ToArray();
 
-        private async Task InsertRowAsync(SpannerCommand writeCommand)
+        private Task InsertRowAsync(SpannerCommand writeCommand)
         {
             string k = UniqueString();
             writeCommand.Parameters["K"].Value = k;
-            _stringValues[k] = (string) (writeCommand.Parameters["StringValue"].Value = GetLargeString(MinDataSize, MaxDataSize));
-            _stringArrayValues[k] = (string[]) (writeCommand.Parameters["StringArrayValue"].Value = GetLargeStringArray());
-            _bytesValues[k] = (byte[]) (writeCommand.Parameters["BytesValue"].Value = GetLargeBytes(MinDataSize, MaxDataSize));
+            _stringValues[k] = (string) (writeCommand.Parameters["StringValue"].Value =
+                GetLargeString(MinDataSize, MaxDataSize));
+            _stringArrayValues[k] = (string[]) (writeCommand.Parameters["StringArrayValue"].Value =
+                GetLargeStringArray());
+            _bytesValues[k] = (byte[]) (writeCommand.Parameters["BytesValue"].Value =
+                GetLargeBytes(MinDataSize, MaxDataSize));
             _bytesArrayValues[k] = (byte[][]) (writeCommand.Parameters["BytesArrayValue"].Value = GetLargeBytesArray());
 
-            await writeCommand.ExecuteNonQueryAsync();
+            return writeCommand.ExecuteNonQueryAsync();
         }
 
         [Fact]
         public async Task TestChunking()
         {
             Logger.Instance.Info($"Seed={_seed}");
-            int rowsRead = 0;
-            int rowsToWrite = _random.Next(5) + 1;
+            var rowsRead = 0;
+            int rowsToWrite = _random.Next(1, 5);
```

Comment:
This is not the same as the original. It produces values in the 1-4 range. Use 6 here or update the comment below.